lib/Syntax/README.md

```diff
@@ -163,8 +163,8 @@ struct YourStruct {}
 At any point in the building process, you can call `build()` and get a
 reasonably formed Syntax node (i.e. with no raw `nullptr`s) using what you've
 provided to the builder so far. Anything that you haven't supplied is marked as
-*missing*. This is essentially what the parser does so, looking forward to
-future adoption, the builders are designed with the parser in mind, with the
```

Comment:
The punctuation here is more correct as written than in the revised version.

Reply:
I strongly disagree. There needs to be a comma between “does” and “so”.

After that it is more stylistic, and in my view reads better as revised.

Reply:
How about: "This is essentially what the parser does, so—looking forward to future adoption—the builders are designed with the parser in mind, with the

Reply:
Good thinking, Harlan, I’ll make it use em-dashes.

Reply:
Commas are more correct; em dashes are used to offset [shifts in thought](http://theweek.com/articles/460264/youre-using-that-dash-wrong):

> It's often a bad idea to use it where you would use a comma — unless the comma is at a clause boundary and you want to signify a greater break.

Reply:
Em dashes are used for [other things](https://en.wikipedia.org/wiki/Dash#Em_dash) as well, and would be perfectly correct here. However, I have reverted them to commas, and placed a semicolon between “does” and “so”, thus breaking up the run-on sentence after its first independent clause.